ext/nio4r/bytebuffer.c

```diff
@@ -75,6 +75,7 @@ void Init_NIO_ByteBuffer()
 static VALUE NIO_ByteBuffer_allocate(VALUE klass)
 {
     struct NIO_ByteBuffer *bytebuffer = (struct NIO_ByteBuffer *)xmalloc(sizeof(struct NIO_ByteBuffer));
+    bytebuffer->buffer = NULL;
```

Comment:
And I thought it was going to be an easy patch :) I'll do it tomorrow as I come back to the laptop.